docs/README.md

```diff
@@ -130,84 +98,78 @@ graph TD
+    )
+    ```
+3. Internally, `__Router_dispatch(...)` and `mailbox.dispatch(...)` are called in `HypERC20Collateral` to dispatch the message to the Mailbox.
+4. The `Mailbox` calls the default Hook (created by Hyperlane) to handle the dispatched message.
 
-**Off chain**
+**Off-chain**
 
-5. Hashi relayer (managed by CCIA team) listen for events from `Yaho` contracts and request the reporter contracts to
-   relay token relaying message.
-6. Hashi executor (managed by CCIA team) listen to event from each Hashi adapter contracts and call
-   `Yaru.executeMessages`. **This step checks whether the Hashi adapters agree on a specify message id** (a threshold
-   number of hash is stored), and set the message Id to verified status.
-7. Validator (run by Hyperlane & LUKSO team) will sign the Merkle root when new dispatches happen in Mailbox.
-8. Hyperlane relayer (run by Hyperlane team) relays the message by calling `Mailbox.process(...)`.
+5. Validator (run by Hyperlane & LUKSO team) signs the Merkle root when new dispatches happen in Mailbox.
+6. Hyperlane relayer (run by Hyperlane team) relays the message by calling `Mailbox.process(...)`.
 
-**on LUKSO chain**
+**On LUKSO chain**
 
-8. When
-   [`Mailbox.process(...)`](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/3d116132b87d36af9576d6b116f31a53d680db4a/solidity/contracts/Mailbox.sol#L188-L197)
-   is called, it will:
-
-- 8.1. check with Multisig ISM (includes Hashi ISM), whether the message is signed by validators & verified by Hashi
-  ISM.
-- 8.2. If so, it will mint [HypLSP7](./src/HypLSP7.sol) tokens to the receiver.
+7. When [`Mailbox.process(...)`](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/3d116132b87d36af9576d6b116f31a53d680db4a/solidity/contracts/Mailbox.sol#L188-L197) is called, it:
+   - 7.1. checks with Multisig ISM whether the message is signed by validators.
+   - 7.2. If so, it mints [HypLSP7](./src/HypLSP7.sol) tokens to the receiver.
 
 ### LUKSO -> Ethereum
 
-> **Note:** This detailed diagram corresponds to the [**scenario 4**](#lukso---ethereum) above. Where an ERC20 token was
-> bridged from Ethereum to LUKSO and we want to bridge back to Ethereum (_e.g: wrapped DAI as HypLSP7_).
-
-![LUKSO to Ethereum bridge flow](../assets/flow-lukso-ethereum-hashi-bridge.png)
-
-**on LUKSO chain**
+> **Note:** This detailed diagram corresponds to the [**scenario 4**](#lukso---ethereum) above. Where an ERC20 token was bridged from Ethereum to LUKSO and we want to bridge back to Ethereum (_e.g.: wrapped DAI as HypLSP7_).
 
-> _Step 1 to 3 needs to be confirmed_
+**On LUKSO chain**
 
-1. User transfers LSP7 token to HypLSP7 contract and the tokens are burnt.
-2. HypLSP7 contract calls `Mailbox` to pass the message.
-3. `Mailbox` calls Default Hook (created by Hyperlane) and Hashi Hook (created by CCIA team).
-4. Hashi Hook dispatches the token relaying message from Yaho contracts.
+1. User approves `HypLSP7` contract to burn the specified amount of tokens.
```

Comment:
This step should be removed. When bridging back, there is no `approve(...)` transaction. Tokens are burnt straight away when calling `transferRemote(...)` on the `HypLSP7` token contract.